omnichannel-server/src/feature/message-source/MessageParser.ts

```diff
@@ -0,0 +1,62 @@
+import { MessageBirdPayload } from "./interfaces/MessageBirdPayload";
+import { MessageTwilioPayload } from "./interfaces/MessageTwilioPayload";
+import { MessagePayload } from "./interfaces/MessagePayload";
+
+export const MessageParser = ({body}) => {
+    const messageBody = <MessageBirdPayload|MessageTwilioPayload>body;
+
+    if(isBird(<MessageBirdPayload>messageBody)) {
+
+        // convert MessageBirdPayload to MessagePayload
+        const MessageBody:MessagePayload = {
+            from: (<MessageBirdPayload>messageBody).contactPhoneNumber,
+            message: (<MessageBirdPayload>messageBody).payload,
+            currentTime: (<MessageBirdPayload>messageBody).currentTime,
+            messageProvider: 'bidr'
+        };
+
+        return MessageBody;
+    } else if( isTwilio(<MessageTwilioPayload>messageBody) ) {
+
+        // convert MessageTwilioPayload to MessagePayload
+        const MessageBody:MessagePayload = {
+            from: (<MessageTwilioPayload>messageBody).Number,
+            message: (<MessageTwilioPayload>messageBody).Message,
+            currentTime: (<MessageTwilioPayload>messageBody).Time,
+            messageProvider: 'twilio'
+        };
+                
+        return MessageBody;
+    }
+    
+}
+
+export const ResponseMessage = (message:MessagePayload) => {
+
+    if(message.messageProvider === 'bidr'){
+        const MessageBody:MessageBirdPayload = {
+            contactPhoneNumber: message.from,
+            payload:message.message,
+            currentTime: message.currentTime,
+        };
+
+        return MessageBody;
+    } else if(message.messageProvider === 'twilio'){
+        const MessageBody:MessageTwilioPayload = {
+            Number: message.from,
+            Message:message.message,
+            Time: message.currentTime,
+        };
+
+        return MessageBody;
+    }
+
+}
+
+const isBird = (message:MessageBirdPayload) => {
+    return message.contactPhoneNumber !== undefined;
+}
+
+const isTwilio = (message: MessageTwilioPayload) => {
+    return message.Number !== undefined;
+}
```

Comment:
Vas por muy buen camino con estas funciones y la idea en general de `MessageParser`. Intenta convertir `MessageParser` a un `class` y que estas funciones sean métodos privados: 

```suggestion
private isBird = (message:MessageBirdPayload): boolean {
    return message.contactPhoneNumber !== undefined;
}

private isTwilio = (message: MessageTwilioPayload): boolean {
    return message.Number !== undefined;
}
```

Reply:
A bueno veré como hacer ese cambio